app/socket-apis/upsert-rank.js

```diff
@@ -0,0 +1,61 @@
+//https://github.com/EnCiv/civil-pursuit/issues/211
+
+const Joi = require('joi')
+const Ranks = require('../models/ranks')
+
+// Define the validation schema using Joi
+const rankSchema = Joi.object({
+  _id: Joi.string().optional(), // _id is optional when creating a new document
+  stage: Joi.string().valid('pre', 'post', 'why').required(), // stage must be one of 'pre', 'post', 'why'
+  category: Joi.string().valid('most', 'least', 'neutral').required(), // category must be one of 'most', 'least', 'neutral'
+  parentId: Joi.string().required(), // parentId is required
+  userId: Joi.string().optional(), // userId is optional and will be deleted later if needed
+  discussionId: Joi.string().required(), // discussionId is required
+  round: Joi.number().integer().min(1).required(), // round must be an integer >= 1
+})
+
+async function validateRankObj(rankObj) {
+  try {
+    // Use Joi to validate the rankObj against the schema
+    await rankSchema.validateAsync(rankObj)
```

Comment:
This works, and it follows the pattern of the post-point-group file.  But I think we need to improve on that. 
Also, we should leave console.log statements in code that goes to production. They are great for debugging, but after it works the need to go. 

without creating a separate function, please just use rankSchema.validate - see upsert-why.js that just got merged into master. 
```js
  const { error } = schema.validate(rankObj);
  if (error) {
    console.error('Validation error in upsert-rank:', error.details[0].message);
    return cb && cb(null); // Return validation error
  }
```
Also, if you would make this change to post-point-groups  too, that would be appreciated.  It's an unnecessary load and more confusing to use async for this.